packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2614,50 +2614,47 @@ export class ContainerRuntime
 	 * @param messageArg - Sequenced message for a distributed document.
 	 * @param local - true if the message was originally generated by the client receiving it.
 	 */
-	public process(messageArg: ISequencedDocumentMessage, local: boolean) {
-		this.verifyNotClosed();
-
-		// Do shallow copy of message, as the processing flow will modify it.
+	public process({ ...messageArg }: ISequencedDocumentMessage, local: boolean) {
```

Comment:
super nit: I'd call it messageCopy.  It's a copy of the message that was passed in as the argument